proposals/immediate-data.md

```diff
@@ -96,30 +95,64 @@ dictionary GPUPipelineLayoutDescriptor
 ```
 `immediateSize`: Size of immediate data range used in pipeline, type is bytes.
 
-NOTE: `immediateSize` = sizeof(variables) + sizeof(paddings). Follow [ Aligment rules ](https://www.w3.org/TR/WGSL/#alignment-and-size) in wgsl spec.
+NOTE: `immediateSize` = sizeof(variables) + sizeof(paddings). Follow [ Alignment rules ](https://www.w3.org/TR/WGSL/#alignment-and-size) in wgsl spec.
 
 NOTE: two pipeline layouts are defined to be “compatible for immediate data” if they were created with identical immediate data byte size. It means immediate data values can share between pipeline layouts that are compatible for immediate data.
 
 NOTE: Immediate data range follow [out-of-bounds access](https://www.w3.org/TR/WGSL/#out-of-bounds-access) rules in wgsl spec.
 
 ## GPUCommandEncoder
 
-Four new functions in `GPUCommandEncoder`.
+One new function in `GPUBindingCommandsMixin`.
 
 ```javascript
 interface mixin GPUBindingCommandsMixin {
-        void setImmediateDataRange(uint32_t rangeOffset, AllowSharedBufferSource data, optional dataOffset, optional size);
+        void setImmediateData(uint32_t rangeOffset, AllowSharedBufferSource data, optional dataOffset, optional size);
 }
 ```
-NOTE: rangeOffset: Offset in bytes into immediate data range to begin writing at. Requires multiple of 4 bytes.
-NOTE: dataOffset: Offset in into data to begin writing from. Given in elements if data is a TypedArray and bytes otherwise.
 
-# Open Questions:
+NOTE: `rangeOffset`: Offset in bytes into immediate data range to begin writing at. Requires multiple of 4 bytes.
+
+NOTE: `dataOffset`: Offset into data to begin writing from. Given in elements if data is a TypedArray and bytes otherwise.
+
+NOTE: The immediate data is stored in an internal slot `[[immediate data]]` in `GPUBindingCommandsMixin`, which is shared across `GPUComputePassEncoder`, `GPURenderPassEncoder`, and `GPURenderBundleEncoder`. See issue [#5117](https://github.com/gpuweb/gpuweb/issues/5117).
```

Comment:
```suggestion
- `rangeOffset`: Offset in bytes into immediate data range to begin writing at. Requires multiple of 4 bytes.
- `dataOffset` and `size` work like in [writeBuffer](https://gpuweb.github.io/gpuweb/#dom-gpuqueue-writebuffer): "Given in elements if data is a TypedArray and bytes otherwise."
- The immediate data is stored in an internal slot `[[immediate data]]` in `GPUBindingCommandsMixin`, which is shared across `GPUComputePassEncoder`, `GPURenderPassEncoder`, and `GPURenderBundleEncoder`. See issue [#5117](https://github.com/gpuweb/gpuweb/issues/5117).
```